### PR TITLE
[Service Connector] `az webapp connection create fabric-sql`: Support Fabric SQL as target

### PR DESCRIPTION
**Pull request by brycechen1849**
**Related command**

`az source connection create fabric-sql`

**Description**

This PR adds a new target service support, Fabric SQL database to the service connector module

**Testing Guide**

**Create linker**

Example: Create linker with user identity

```sh
az webapp connection create fabric-sql 
--source-id /subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Web/sites/{app_name}
--target-id https://api.fabric.microsoft.com/v1/workspaces/{workspace_id}/SqlDatabases/{sql_id}
--user-identity client-id=3af94a55-9cf5-4e70-8616-c540a08a8374 subs-id=6cd159c1-0ef9-41bc-85b1-e8db1a626465 
--client-type dotnet 
--connstr-props 'Server=tcp:renzo-srv-6ae35870-c362-44b9-8389-ada214a46bb5-51240650dd56.database.windows.net,1433' 'Database=AzureServiceConnectorTestSqlDb-4fdf6efe-23a9-4d74-8c4a-4ecc70c4d323'
```

Example: Create linker with system identity

```bash
az webapp connection create fabric-sql 
--source-id /subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Web/sites/{app_name}
--target-id https://api.fabric.microsoft.com/v1/workspaces/{workspace_id}/SqlDatabases/{sql_id}
--system-identity 
--client-type dotnet 
--connstr-props 'Server=tcp:renzo-srv-6ae35870-c362-44b9-8389-ada214a46bb5-51240650dd56.database.windows.net,1433' 'Database=AzureServiceConnectorTestSqlDb-4fdf6efe-23a9-4d74-8c4a-4ecc70c4d323'
```

**Update linker**

update linker with user identity

```bash
az webapp connection create fabric-sql 
--connection fabric_sql_a973e 
--source-id /subscriptions/{sub}/resourceGroups/{rg}/providers/Microsoft.Web/sites/{app_name}
--target-id https://api.fabric.microsoft.com/v1/workspaces/{workspace_id}/SqlDatabases/{sql_id}
--user-identity client-id=3af94a55-9cf5-4e70-8616-c540a08a8374 subs-id=6cd159c1-0ef9-41bc-85b1-e8db1a626465 
--client-type dotnet 
--connstr-props 'Server=tcp:renzo-srv-6ae35870-c362-44b9-8389-ada214a46bb5-51240650dd56.database.windows.net,1433' 'Database=AzureServiceConnectorTestSqlDb-4fdf6efe-23a9-4d74-8c4a-4ecc70c4d323'
```

**History Notes**

[Service Connector] `az webapp connection create fabric-sql`: Add fabric sql target support

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Service Connector

**Comment by brycechen1849** [comment]
passwordless experience change is in PR https://github.com/Azure/azure-cli-extensions/pull/7940

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 1]
```diff
@@ -498,8 +498,9 @@ def connection_create_func(cmd, client,  # pylint: disable=too-many-locals,too-m
 
     validate_service_state(parameters)
     if enable_mi_for_db_linker and auth_action != 'optOutAllAuth':
+        logger.warning('Start creating mi for db linker')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:501]: why the warning here?

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 2]
```diff
@@ -1294,3 +1313,26 @@ def connection_update_kafka(cmd, client,  # pylint: disable=too-many-locals
     return client.begin_create_or_update(resource_uri=source_id,
                                          linker_name=connection_name,
                                          parameters=parameters)
+
+
+{
+  "properties": {
+    "targetService": {
+      "type": "ConfluentBootstrapServer",
```
[the remark this answers is not in the document]
Comment (brycechen1849) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py]: removed unused snippet

File: src/azure-cli/azure/cli/command_modules/serviceconnector/action.py [hunk 3]
```diff
@@ -24,6 +24,23 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
...
+        action = self.get_action(values, option_string)
+        namespace.connstr_props = action
+
+    def get_action(self, values, option_string):  # pylint: disable=no-self-use
+        try:
+            properties = defaultdict(list)
+            for (k, v) in (x.split('=', 1) for x in values):
+                properties[k] = v
+            properties = dict(properties)
+            print(properties.keys(), properties.values(), properties.items())
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/action.py:38]: remove debug info

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 4]
```diff
@@ -507,8 +520,9 @@ def connection_create_func(cmd, client,  # pylint: disable=too-many-locals,too-m
 
     validate_service_state(parameters)
     if enable_mi_for_db_linker and auth_action != 'optOutAllAuth':
+        logger.warning('Start creating mi for db linker')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:523]: if the message is intended to show to user, pls make it official and professional,  Start creating managed identity for fabric sql .blahblah...

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py [hunk 5]
```diff
@@ -518,7 +532,7 @@ def connection_create_func(cmd, client,  # pylint: disable=too-many-locals,too-m
         logger.warning(springboot_migration_warning(require_update=False,
                                                     check_version=(not isSecretType),
                                                     both_version=isSecretType))
-
+    logger.warning('request payload: %s', parameters)
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py:535]: pls don't log request payload

**Review by yungezz — CHANGES_REQUESTED** [review]

**Comment by yungezz** [comment]
pls add test to cover the feature

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (brycechen1849) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py]: removed

Comment (brycechen1849) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py]: removed

**Comment by brycechen1849** [comment]
> pls add test to cover the feature

Add test case in extension repo of passwordless https://github.com/Azure/azure-cli-extensions/pull/7940/files#diff-aa352ccc303233df198de786b8a0f24a7b598ff32918aec91bacaf6c524329b7

File: src/azure-cli/azure/cli/command_modules/serviceconnector/action.py (hunk 3 above)
Comment (brycechen1849) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/action.py]: removed

File: src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py (hunk 4 above)
Comment (brycechen1849) [reply src/azure-cli/azure/cli/command_modules/serviceconnector/custom.py]: removed as this is not necessary information

**Comment by brycechen1849** [comment]
Hi @kairu-ms, can you please proceed to merge the PR

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**